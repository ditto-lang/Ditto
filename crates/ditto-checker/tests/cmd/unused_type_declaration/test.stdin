module Test exports (A(..));

type A = A;
type B = B;
