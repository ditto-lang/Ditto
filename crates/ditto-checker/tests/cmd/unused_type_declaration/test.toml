bin.name = "ditto-checker-testbin-check-module"
args = ["unused_type_declaration.ditto"]
fs.sandbox = true
