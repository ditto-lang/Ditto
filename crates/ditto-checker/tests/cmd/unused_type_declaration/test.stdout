{
  "module_name": [
    "Test"
  ],
  "exports": {
    "types": {
      "A": {
        "Type": {
          "doc_comments": [],
          "doc_position": 0,
          "kind": "Type"
        }
      }
    },
    "constructors": {
      "A": {
        "doc_comments": [],
        "doc_position": 0,
        "constructor_type": {
          "type": "Constructor",
          "data": {
            "constructor_kind": "Type",
            "canonical_value": {
              "module_name": [
                null,
                [
                  "Test"
                ]
              ],
              "value": "A"
            },
            "source_value": {
              "module_name": null,
              "value": "A"
            }
          }
        },
        "return_type_name": "A"
      }
    },
    "values": {}
  },
  "types": {
    "B": {
      "Type": {
        "doc_comments": [],
        "type_name_span": {
          "start_offset": 47,
          "end_offset": 48
        },
        "kind": "Type"
      }
    },
    "A": {
      "Type": {
        "doc_comments": [],
        "type_name_span": {
          "start_offset": 35,
          "end_offset": 36
        },
        "kind": "Type"
      }
    }
  },
  "constructors": {
    "B": {
      "doc_comments": [],
      "doc_position": 0,
      "constructor_name_span": {
        "start_offset": 51,
        "end_offset": 52
      },
      "fields": [],
      "return_type": {
        "type": "Constructor",
        "data": {
          "constructor_kind": "Type",
          "canonical_value": {
            "module_name": [
              null,
              [
                "Test"
              ]
            ],
            "value": "B"
          },
          "source_value": {
            "module_name": null,
            "value": "B"
          }
        }
      },
      "return_type_name": "B"
    },
    "A": {
      "doc_comments": [],
      "doc_position": 0,
      "constructor_name_span": {
        "start_offset": 39,
        "end_offset": 40
      },
      "fields": [],
      "return_type": {
        "type": "Constructor",
        "data": {
          "constructor_kind": "Type",
          "canonical_value": {
            "module_name": [
              null,
              [
                "Test"
              ]
            ],
            "value": "A"
          },
          "source_value": {
            "module_name": null,
            "value": "A"
          }
        }
      },
      "return_type_name": "A"
    }
  },
  "values": {},
  "values_toposort": []
}